### PR TITLE
Add NodePort to remainder of services

### LINKED ISSUE #297: feat: Finish NodePort assignment for remainder of services
**Is this a bug report or feature request?** (choose one):
Feature request

**Development or Deployment Environment?**:
Both
**Release Tag or Master**:
0.3.0/master

**Expected Behavior**:
Services should have a boolean to expose a NodePort for the service.

**Any Additional Comments**:
A standardized template for how we want to format the values.yaml has been decided [here](https://github.com/att-comdev/openstack-helm/pull/274).  NodePorts for the rest of the OpenStack services other than Heat need to be created.

Applies to the following services: cinder-api, glance-api, keystone-api, neutron-server, nova-metadata, nova-osapi

### PR DESCRIPTION
**Pull request by alraddarla**
**What is the purpose of this pull request?**: 
Allows a NodePort for each service to be exposed.

**What issue does this pull request address?**: Fixes #297 

**Notes for reviewers to consider**:
After speaking with Brandon, he said we would want to mimic what the Heat files look like. This is why there are changes in the network section of the values file for each service on top of adding the NodePort boolean.

**Specific reviewers for pull request**:
@alanmeadows @v1k0d3n @larryrensing @intlabs @wilkers-steve

### CODE REVIEW

File: glance/values.yaml [hunk 1]
```diff
@@ -64,9 +64,15 @@ keystone:
...
-    registry: 9191
+  api:
+    name: "glance-api"
+    port: 9292
+    node_port:
+      enabled: false
+      port: 30092
+  registry:
+    name: "glance-registry"
+    port: 9191
```
Comment (intlabs) [inline glance/values.yaml]: Could you also setup nodeports for glance registry.

File: keystone/templates/service.yaml [hunk 2]
```diff
@@ -19,8 +19,14 @@ metadata:
...
   ports:
   - name: keystone-api-public
-    port: {{ .Values.network.port.api }}
+    port: {{ .Values.network.api.port }}
+    {{ if .Values.network.api.node_port.enabled }}
+      nodePort: {{ .Values.network.api.node_port.port }}
+    {{ end }}
   - name: keystone-api-admin
-    port: {{ .Values.network.port.admin }}
+    port: {{ .Values.network.admin.port }}
```
Comment (intlabs) [inline keystone/templates/service.yaml]: The admin endpoint will also want a nodeport.

File: neutron/values.yaml [hunk 3]
```diff
@@ -78,9 +78,15 @@ network:
...
-    metadata: 8775
+  server:
+    name: "neutron-server"
+    port: 9696
+    node_port:
+      enabled: false
+      port: 30096
+  metadata:
+    name: "neutron-metadata"
+    port: 8775
```
Comment (intlabs) [inline neutron/values.yaml]: The metadata agent will need a nodeport.

File: nova/values.yaml [hunk 4]
```diff
@@ -61,11 +61,21 @@ network:
...
+      port: 30774
+  metadata:
+    name: "nova-metadata"
+    port: 8775
+    node_port:
+      enabled: false
+      port: 30775
+  novncproxy:
+    name: "nova-novncproxy"
+    port: 6080
```
Comment (intlabs) [inline nova/values.yaml]: The NoVNCproxy will also want a nodeport.

File: neutron/values.yaml (hunk 3 above)
Comment (intlabs) [inline neutron/values.yaml]: @alraddarla As discussed on Slack: we shouldn't be exposing this publicly as it only operates on tenant networks. Sorry for sending you on a wild goose chase :/

**Comment by wilkers-steve** [comment]
This looks good @alraddarla.  I can't see anything else beyond @intlabs's comments.  Once those get addressed, this should be good to go.

File: glance/values.yaml (hunk 1 above)
Comment (alraddarla) [inline glance/values.yaml]: Done.

File: keystone/templates/service.yaml (hunk 2 above)
Comment (alraddarla) [inline keystone/templates/service.yaml]: Done.

File: neutron/values.yaml (hunk 3 above)
Comment (alraddarla) [inline neutron/values.yaml]: Pretty rude...

**Comment by intlabs** [comment]
LGTM :+1:

**Comment by alraddarla** [comment]
DO NOT MERGE --- this crashes keystone currently

File: keystone/templates/service.yaml [hunk 5]
```diff
@@ -19,8 +19,20 @@ metadata:
...
+    {{ if .Values.network.admin.node_port.enabled }}
+      nodePort: {{ .Values.network.admin.node_port.port }}
+    {{ end }}
   selector:
     app: keystone-api
+  selector:
+    app: keystone-api
```
Comment (wilkers-steve) [inline keystone/templates/service.yaml]: Here's what's crashing keystone for you.  Sorry I missed this earlier.

Comment (wilkers-steve) [inline keystone/templates/service.yaml]: The line below this actually. The "or" operand

File: neutron/values.yaml (hunk 3 above)
Comment (alanmeadows) [inline neutron/values.yaml]: Not nearly as bad as crashing keystone.

**Comment by alanmeadows** [comment]
@alraddarla This is a great enhancement with a lot of good consistency cleanup embedded in this.  The project needs this kind of attention across a large number of areas. Nice work. The only issue is the `{{ if .. or  .. }}` cleanup @wilkers-steve pointed out, but otherwise looks good to me.

**Pull request merged**